src/Providers/Stripe/Customer/Service.php

```diff
@@ -89,29 +89,37 @@ public function addCard(Customer $customer, Card $card): Card
     /**
      * Deletes a customers card using the Stripe SDK.
      *
-     * @param Customer $customer
-     * @param Card $card
+     * @param string $customerUid
+     * @param string $cardUid
      *
      * @return void
      */
-    public function deleteCard(Customer $customer, Card $card): void
+    public function deleteCard(string $customerUid, string $cardUid): void
     {
-        StripeCustomer::deleteSource($customer->uid, $card->uid);
+        $cardToken = new StripeToken($cardUid);
+
+        if ($cardToken->type === StripeToken::PAYMENT_METHOD) {
```

Comment:
could be
```php
if ((new StripeToken($cardUid))->type === StripeToken::PAYMENT_METHOD) {
```
or
```php
if ((new StripeToken($cardUid))->is(StripeToken::PAYMENT_METHOD)) {
```
or
```php
if ((new StripeToken($cardUid))->isPaymentMethod()) {
```

Reply:
I'm also going to update it into a switch so that it's easier to read.